pkg/pullrequest/api.go

```diff
@@ -272,15 +272,22 @@ func (h *Handler) uploadStatuses(ctx context.Context, statuses []*scm.Status, sh
 	}
 
 	// Index the statuses so we can avoid sending them if they already exist.
-	csMap := map[string]scm.State{}
+	csMap := map[string]scm.Status{}
 	for _, s := range cs {
-		csMap[s.Label] = s.State
+		// Statuses are returned by the API in reverse chronological order:
+		// https://developer.github.com/v3/repos/statuses/#list-statuses-for-a-specific-ref
+		// We only consider the most recent status with a give label.
+		if _, ok := csMap[s.Label]; ok {
+			continue
+		}
+		//
+		csMap[s.Label] = *s
 	}
 
 	var merr error
 
 	for _, s := range statuses {
-		if csMap[s.Label] == s.State {
+		if csMap[s.Label].State == s.State && csMap[s.Label].Target == s.Target {
```

Comment:
IIUC, this is to allow updates of statuses event if their state is the same. Do we also want to compare the description for the same reason?

Reply:
I care about the target because that changes between runs, but yeah, I will add description too

Reply:
done